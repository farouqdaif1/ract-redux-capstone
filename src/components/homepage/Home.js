import React from 'react';
import { useSelector } from 'react-redux';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Details from '../detailspage/Details';
import Nav from './Nav';
import CITYS from './CITYS';

const Home = () => {
  const city = useSelector((state) => state.homeReducer);
  return (
    <div className="container-home">
      <Router>
        <Nav />
        <Routes>
          <Route exact path="/" element={<CITYS />} />
          <Route path="/details" element={<Details city={city} />} />
        </Routes>
      </Router>
    </div>

  );
};
export default Home;
