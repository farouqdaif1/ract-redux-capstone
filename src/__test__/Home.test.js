import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Home from '../components/homepage/Home';

describe('Render components', () => {
  it('render HomePage', () => {
    const tree = renderer
      .create(<Provider store={store}><Home /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
