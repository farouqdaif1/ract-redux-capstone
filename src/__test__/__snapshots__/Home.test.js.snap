// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render HomePage 1`] = `
<div
  className="container-home"
>
  <div
    className="Navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="arrow"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </a>
    <div
      className="log"
    >
       Weather In Egypt
    </div>
    <input
      className="search"
      placeholder="Search.."
      type="text"
    />
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 12V6c0-2.217-1.785-4.021-3.979-4.021a.933.933 0 0 0-.209.025A4.006 4.006 0 0 0 8 6v6c0 2.206 1.794 4 4 4s4-1.794 4-4zm-6 0V6c0-1.103.897-2 2-2a.89.89 0 0 0 .163-.015C13.188 4.06 14 4.935 14 6v6c0 1.103-.897 2-2 2s-2-.897-2-2z"
      />
      <path
        d="M6 12H4c0 4.072 3.061 7.436 7 7.931V22h2v-2.069c3.939-.495 7-3.858 7-7.931h-2c0 3.309-2.691 6-6 6s-6-2.691-6-6z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
      />
    </svg>
  </div>
  <div
    className="container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="map"
      >
        <img
          alt="egypt"
          src="resize.jpg"
        />
      </div>
    </a>
    <div
      className="buttons"
    >
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Cairo"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Cairo
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Alexandria"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Alexandria
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Aswan"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Aswan
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Asyut"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Asyut
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Damietta"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Damietta
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Giza"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Giza
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Ismailia"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Ismailia
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Luxor"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Luxor
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Suez"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Suez
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Sohag"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Sohag
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Qena"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Qena
        </div>
      </a>
      <a
        className="link"
        href="/details"
        onClick={[Function]}
      >
        <div
          className="town"
          id="Minya"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Minya
        </div>
      </a>
    </div>
  </div>
</div>
`;
